src/com/linkedin/parseq/TaskDescriptor.java

```diff
@@ -0,0 +1,20 @@
+package com.linkedin.parseq;
+
+/**
+ * An API to provide description for task.
+ *
+ * parseq-lambda-names provide an implementation for this interface using Java SPI. Any changes made to this interface
+ * might require a change to implementation in parseq-lambda-names.
+ *
+ * @author Siddharth Sodhani (ssodhani@linkedin.com)
+ */
+public interface TaskDescriptor {
+
+  /**
+   * Given class which could correspond to generated lambda expressions etc infer appropriate description for it
+   *
+   * @param clazz
+   * @return description for task
+   */
+  String getDescription(Class<?> clazz);
```

Comment:
It is true that the input to this method eventually will always be a Class and it is true for Task interface where we want to generate better task description.
What I am thinking is that this could be useful in potentially different scenarios. One example is a stack trace. Currently stack trace contains lambda class name. It might be possible to improve it by substituting it with description provided by this class. In this scenario we would have only a class name as a String and converting it to Class for the purpose of this interface could be time consuming.
Do we use the Class object passed to this method in any way except for calling toString() on it?